packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -478,6 +478,41 @@ class SummarizerNodeWithGC extends SummarizerNode implements IRootSummarizerNode
 		return child;
 	}
 
+	/**
+	 * Updates the state of the child if required. For example, if a summary is currently being  tracked, the child's
+	 * summary tracking state needs to be updated too.
+	 * Also, in case a child node gets realized in between Summary Op and Summary Ack, let's initialize the child's
+	 * pending summary as well. Finally, if the pendingSummaries entries have serializedRoutes, replicate them to the
+	 * pendingSummaries from the child nodes.
+	 * @param child - The child node whose state is to be updated.
+	 */
+	protected maybeUpdateChildState(child: SummarizerNodeWithGC) {
+		// If a summary is in progress, this child was created after the summary started. So, we need to update the
+		// child's summary state as well.
+		if (this.isSummaryInProgress()) {
```

Comment:
SummarizerNodeWithGC is an extension of SummarizerNode - it should only add functionality not duplicate it otherwise things can become out of sync resulting in bugs.

Here, it must call `super.maybeUpdateChildState()` to let the base class run its logic and not duplicate the base class logic. If the base class logic changes, then this will become stale.
So, it should first call `super.maybeUpdateChildState()` and let it create pending summaries for the child. Then it should iterative over those pending summaries and update it to be SummaryNodeWithGC.

Reply:
Yes, I know it is an extension - the problem is when SummarizerNode adds to the list of pending summaries, it does not add as a SummarizerNodeWithGC so I would need to re-create the nodes and I thought it would be a waste of time. Anyways, I can do the super then process the pending summaries again.

Reply:
Yeah, its annoying :-(. Maybe we can create the summarizer node with GC nodes first and then pass it to summarizer node so it can add its info? But then it will have to be changed in the CompleteSummaryCore method as well.